packages/block-editor/src/components/link-control/index.js

```diff
@@ -194,14 +205,37 @@ function LinkControl( {
 		// If it's potentially a URL search then concat on a URL search suggestion
 		// just for good measure. That way once the actual results run out we always
 		// have a URL option to fallback on.
-		return couldBeURL && ! args.isInitialSuggestions
-			? results[ 0 ].concat( results[ 1 ] )
-			: results[ 0 ];
+		results =
+			couldBeURL && ! args.isInitialSuggestions
+				? results[ 0 ].concat( results[ 1 ] )
+				: results[ 0 ];
+
+		// Here we append a faux suggestion to represent a "CREATE" option. This
+		// is detected in the rendering of the search results and handeled as a
+		// special case. This is currently necessary because the suggestions
+		// dropdown will only appear if there are valid suggestions and
+		// therefore unless the create option is a suggestion it will not
+		// display in scenarios where there are no results returned from the
+		// API. In addition promoting CREATE to a first class suggestion affords
+		// the a11y benefits afforded by `URLInput` to all suggestions (eg:
+		// keyboard handling, ARIA roles...etc).
+		// Note also that due to the flow of data and the nature of the
+		// controlled components the value of the `url` property must correspond
+		// to the text value of the `<input>` otherwise it will result in an
+		// incorrectly named entity being created.
+		return results.concat( {
+			id: uniqueId(),
```

Comment:
@aduth Again is this a good idea? See comment about the other usage of `uniqueId` in this component.

Reply:
Being discussed [here](https://github.com/WordPress/gutenberg/pull/19775#discussion_r374766073).